Language_Parser/actionVerbs.py

```diff
@@ -135,18 +160,40 @@ def drop(info):
     game = info["Game"]
     player = game.getPlayer()
     room = player.getLocation()
-    for item in info["Items"]:
-        if item == "polaroid":
-            item = identifyPolaroid(player)
-            if item is None:
-                print("You don't have that.")
-                return
-        for possession in player.getInventory():
-            if possession == item:
-                print(possession.verbResponses("Drop"))
-                player.removeInventory(item)
-                room.addDroppedItem(possession)
-                return
+    item = info["Items"]
+    allItems = player.getInventory() + room.getAccessibleItems()
+
+    # Try to pick up the item
+    if "polaroid" in item:
+        item = identifyPolaroid(player)
+        if item is None:
+            print(errorString)
+            return
+    for possession in player.getInventory():
+        if "flashlight" in item:
+            item = getSteppedItemName(allItems, item)
```

Comment:
I think the "if "flashlight" in item" portion could be moved outside the for loop, since it only needs to happen once

Reply:
Agreed. Done.